### PR TITLE
Reconcile with solr_bulk repo

### PR DESCRIPTION
**Pull request by pcm32**
We found that major aspects of the bulk index analytics collection schema where in https://github.com/ebi-gene-expression-group/solr-bulk instead of here. This PR aims to bring those differences towards archiving that older repo.

### CODE REVIEW

**Comment by pcm32** [comment]
Not sure how this happened based on https://github.com/ebi-gene-expression-group/index-gxa/pull/3...

File: bin/create-gxa-analytics-schema.sh [hunk 1]
```diff
@@ -27,6 +27,101 @@ curl -X POST -H 'Content-type:application/json' --data-binary '{
...
+  "add-field":
+  {
+    "name": "is_private",
+    "type": "boolean"
+  }
+}' http://${HOST}/solr/${CORE}/schema
+
+#############################################################################################
+
+printf "\n\nDelete copy field rule bioentity_identifier -> bioentity_identifier_search\n"
```
Comment (upendrakumbham) [inline bin/create-gxa-analytics-schema.sh:49]: Please move this `delete copy field rule `snippet to before `add copy field rule` and similarly align `bioentity_identifier`, ``bioentity_identifier_search``field order as well.

File: bin/create-gxa-analytics-schema.sh [hunk 2]
```diff
@@ -27,6 +27,101 @@ curl -X POST -H 'Content-type:application/json' --data-binary '{
...
+  {
+    "name": "assay_group_id",
+    "type": "string",
+    "docValues": true
+  }
+}' http://${HOST}/solr/${CORE}/schema
+
+#############################################################################################
+
+# TODO Do we need assay? is not in https://github.com/ebi-gene-expression-group/solr-bulk/blob/master/bin/create-bulk-analytics-schema.sh
```
Comment (upendrakumbham) [inline bin/create-gxa-analytics-schema.sh:124]: @alfonsomunozpomer - can you please confirm this? I'm not sure about this.

File: bin/create-gxa-analytics-schema.sh [hunk 3]
```diff
@@ -269,6 +734,37 @@ curl -X POST -H 'Content-type:application/json' --data-binary '{
...
+    "signatureField": "id",
+    "overwriteDupes": "true",
+    "fields": "experiment_accession,bioentity_identifier,assay_group_id,contrast_id",
+    "signatureClass": "solr.processor.Lookup3Signature"
+  }
+}' http://${HOST}/solr/${CORE}/config
+
+############################################################################
+
+# TODO do we need all the ones below for bulk analytics? They are not in https://github.com/ebi-gene-expression-group/solr-bulk/blob/master/bin/create-bulk-analytics-schema.sh
```
Comment (upendrakumbham) [inline bin/create-gxa-analytics-schema.sh:767]: @alfonsomunozpomer -can you please confirm this?

**Comment by alfonsomunozpomer** [comment]
I think trying to reconcile the two repositories isn’t the best solution. This repository’s original goal was to move bulk forward and have the same data architecture as in single-cell after all the lessons we learned. In fact, if you read https://github.com/ebi-gene-expression-group/index-gxa/pull/3:

> Bulk analytics schema is created like the one for single cell.

I think that it’s better to have scripts such as `create-bulk-analytics...` to differentiate them from `create-gxa-analytics...` which follow the same naming scheme as in SC.

**Comment by pcm32** [comment]
The original idea of this repo was to have a testable setup, like what we have for SCXA. I'm all for keeping what we learned in terms of schema, but the original intent of what I asked to Suhaib was to have something testable for bulk indexing. Given the document structure, I presume there is no issue in keeping newer and older fields so that we can make the transition, but I would like to have everything here since this is where we are testing it, both with the loading scripts and with the atlas-web-cli. I have also made the work for moving it into Solr 8.7 here.

File: bin/create-gxa-analytics-schema.sh (hunk 1 above)
Comment (pcm32) [reply bin/create-gxa-analytics-schema.sh:49]: I think that there is a dependency between fields that means that copy field rule might need to be deleted before you can delete bioentity_identifier, and in turn the copy field action needs to be created after the field is copying (bioentitiy_identifier). The same with the search. This is as well the same order that it was in the source repo (I didn't change the order).

File: bin/create-gxa-analytics-schema.sh (hunk 2 above)
Comment (pcm32) [reply bin/create-gxa-analytics-schema.sh:124]: We can as well leave these as a transition, and start filling both.

File: bin/create-gxa-analytics-schema.sh (hunk 1 above)
Comment (upendrakumbham) [reply bin/create-gxa-analytics-schema.sh:49]: Yes, you're right @pcm32.Thanks.

**Review by upendrakumbham — APPROVED** [review]
All good to me!

**Comment by pcm32** [comment]
Dumping this in favour of adding the scripts side by side in #18

**Pull request closed without merging**